### ISSUE TITLE
Downloading archive failes with "406 Not Acceptable"

### ISSUE THREAD

**Issue by ktaube** [jdalrymple/gitbeaker#3248]
**Description**

- Node.js version: v18.16.0
- Gitbeaker version: ^38.3.0
- Gitbeaker release (cli, node, browser, core, requester-utils): rest
- OS & version: MacOS & AWS Lambda runtime

Calling `Gitlab.Repositories.showArchive` results in Gitlab responding `406 Not Acceptable` error.
Trying the same call using curl or `node-fetch` works successfully.

**Steps to reproduce**

Run following code example to reproduce:
```js
// gibeaker-braker.js
const { Gitlab } = require('@gitbeaker/rest');

const yourAccessToken = '...';
const projectId = 0;
const commitSha = 'main';

const api = new Gitlab({
  oauthToken: yourAccessToken,
  host: 'https://gitlab.com',
});

(async () => {
  // This errors out with "406 Not Acceptable"
  const archive = await api.Repositories.showArchive(projectId, {
    sha: 'main',
    fileType: 'zip',
  });

  console.log({ archive });
})();
```

**Expected behaviour**

Should return successful blob response.

**Actual behaviour**

Fails with `406 Not Acceptable`

**Possible fixes**

So what I ended up doing is just implementing this call myself:
```js
  const yourAccessToken = '...';
  const projectId = 0;

  const url = `https://gitlab.com/api/v4/projects/${projectId}/repository/archive.zip`;

  const response = await fetch(url, {
    headers: {
      Authorization: `Bearer ${yourAccessToken}`,
    },
  });

  // successfully returns the blob data
  const blob = await response.blob();
```

**Checklist**
- [x] I have checked that this is not a duplicate issue.
- [x] I have read the documentation.

**Comment by jdalrymple** [comment]
This is related to #3232 . By switching from same-origin to cors it bombs out. -_-

**Comment by jdalrymple** [comment]
Still an issue, trying to figure it out: https://stackoverflow.com/questions/76236958/how-to-use-nodejs-browsers-native-fetch-library-to-download-a-gitlab-archive-wit

**Comment by jdalrymple** [comment]
The cause is due to the cors mode set by the native fetch library. AFAIK, there is no way to override this atm which is highly frustrating. Im looking into alternatives

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`38.4.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/38.4.0) :rocket:

**Comment by jdalrymple** [comment]
:rocket: Issue was released in [`38.4.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/38.4.0) :rocket:
